app/views/warehouses/index.html.erb

```diff
@@ -12,20 +12,22 @@
   </div>
  <div class="col-sm-8">
     <div class="title-action">
-      <%= link_to 'New Warehouse', hubs_path, :class => "btn btn-primary"%>
+      <%= link_to 'New Warehouse',  new_hub_warehouse_path(0), :class => "btn btn-primary"%>
```

Comment:
yap. unless we remove the hierarchy, it doesn't accept warehouses with hubs. i set that in the controller for requests directly from warehouses page